EIPS/EIPs/eip-5075.md

```diff
@@ -0,0 +1,236 @@
+    function transferL(
+        address to,
+        uint256 amount,
+        address token
+    ) public returns (bool success) {
+        TokenInfo storage info = infoL[token];
+
+        if (whitelisted[msg.sender] == 1) {
+            if (address(token) == address(0)) {
+                payable(to).transfer(amount);
+                return(true);
+            } else {
+                return(IERC20(token).transfer(to, amount));
+            }
+        }
+        //used if the asset is ETH and not an ERC20 token
+        else {
+            if (address(token) == address(0)) {
+                //used to get around solidity 0.8 reverts
+                unchecked {
+                    //checks to see if the last transaction was outside the time window to track outflow for limiting
+                    if (timeLimit <= block.timestamp - info.timestamp) {
+                        info.rate = 0;
+                    }
+                    //if the last transaction was within the time window, decreases the tracked outflow rate relative to the time elapsed, so that the limit is able to update in realtime rather than in blocks, making flows smooth, and increasing the rate available as time increases without a transaction
+                    else {
+                        info.rate -= uint224(
+                            (address(this).balance * rateLimit) /
+                                (timeLimit /
+                                    (block.timestamp - info.timestamp)) /
+                                10000
+                        );
+                    }
+                }
+                //increases the tracked rate for the current time window by the amount sent out
+                info.rate += uint224(amount);
+                //revert if the outflow exceeds rate limit
+                require(
+                    info.rate <= (rateLimit * address(this).balance) / 10000
+                );
+                //sets the current time as the last transfer for the token
+                info.timestamp = uint32(block.timestamp);
+                //transfers out
+                payable(to).transfer(amount);
+                return(true);
+                //if the token is a ERC20 token
+            } else {
+                //used to get around solidity 0.8 reverts
+                unchecked {
+                    //checks to see if the last transaction was outside the time window to track outflow for limiting
+                    if (timeLimit <= block.timestamp - info.timestamp) {
+                        info.rate = 0;
+                    }
+                    //if the last transaction was within the time window, decreases the tracked outflow rate relative to the time elapsed, so that the limit is able to update in realtime rather than in blocks, making flows smooth, and increasing the rate available as time increases without a transaction
+                    else {
+                        info.rate -= uint224(
```

Comment:
similar concern about revert due to undeflow due to balance or settings changes

Reply:
Will do